maintenancewindow.go

```diff
@@ -0,0 +1,143 @@
+package wavefront
+
+import "fmt"
+
+const (
+	maintenanceWindowEndpoint = "/api/v2/maintenancewindow"
+)
+
+// MaintenanceWindow represents a maintenance window in Wavefront
+type MaintenanceWindow struct {
+	ID                              string   `json:"id"`
+	RunningState                    string   `json:"runningState"`
+	SortAttr                        int      `json:"sortAttr"`
+	Reason                          string   `json:"reason"`
+	CustomerId                      string   `json:"customerId"`
+	RelevantCustomerTags            []string `json:"relevantCustomerTags"`
+	Title                           string   `json:"title"`
+	StartTimeInSeconds              int64    `json:"startTimeInSeconds"`
+	EndTimeInSeconds                int64    `json:"endTimeInSeconds"`
+	RelevantHostTags                []string `json:"relevantHostTags"`
+	RelevantHostNames               []string `json:"relevantHostNames"`
+	CreatorId                       string   `json:"creatorId"`
+	UpdaterId                       string   `json:"updaterId"`
+	CreatedEpochMillis              int64    `json:"createdEpochMillis"`
+	UpdatedEpochMillis              int64    `json:"updatedEpochMillis"`
+	RelevantHostTagsAnded           bool     `json:"relevantHostTagsAnded"`
+	HostTagGroupHostNamesGroupAnded bool     `json:"hostTagGroupHostNamesGroupAnded"`
+	EventName                       string   `json:"eventName"`
+}
+
+// Options returns the current options for this maintenance window
+func (m *MaintenanceWindow) Options() *MaintenanceWindowOptions {
+	return &MaintenanceWindowOptions{
+		Reason:               m.Reason,
+		Title:                m.Title,
+		StartTimeInSeconds:   m.StartTimeInSeconds,
+		EndTimeInSeconds:     m.EndTimeInSeconds,
+		RelevantCustomerTags: m.RelevantCustomerTags,
+		RelevantHostTags:     m.RelevantHostTags,
+		RelevantHostNames:    m.RelevantHostNames,
+	}
+}
+
+// MaintenanceWindowOptions represents the configurable options for a
+// maintenance window.
+type MaintenanceWindowOptions struct {
+	// Required. The reason for the maintenance window.
+	Reason string `json:"reason,omitempty"`
+
+	// Required. The title of the maintenance window.
+	Title string `json:"title,omitempty"`
+
+	// Required. The start time of the maintenance window in seconds since 1 Jan 1970
+	StartTimeInSeconds int64 `json:"startTimeInSeconds,omitempty"`
+
+	// Required. The end time of the maintenance window in seconds since 1 Jan 1970
+	EndTimeInSeconds int64 `json:"endTimeInSeconds,omitempty"`
+
+	RelevantCustomerTags []string `json:"relevantCustomerTags"`
+	RelevantHostTags     []string `json:"relevantHostTags,omitempty"`
+	RelevantHostNames    []string `json:"relevantHostNames,omitempty"`
```

Comment:
two more options that are configurable are relevantHostTagsAnded and hostTagGroupHostNameGroupAnded.

Reply:
Thank you for pointing this out. Do you know of a way I can glean from the JAVA code all the options that are available?  I've been relying on the REST documentation, but I have been finding that the REST documentation doesn't always get updated.

Reply:
Done.